bindings/pyroot/test/rdataframe_asnumpy.py

```diff
@@ -144,24 +144,24 @@ def test_define_columns(self):
         npy = df.AsNumpy(columns=["x", "y"])
         ref = {"x": np.array([1] * 4), "y": np.array([2] * 4)}
         self.assertTrue(sorted(["x", "y"]) == sorted(npy.keys()))
-        self.assertTrue(any(ref["x"] == npy["x"]))
-        self.assertTrue(any(ref["y"] == npy["y"]))
+        self.assertTrue(all(ref["x"] == npy["x"]))
+        self.assertTrue(all(ref["y"] == npy["y"]))
 
     def test_exclude_columns(self):
         df = ROOT.RDataFrame(4).Define("x", "1").Define("y", "2").Define(
             "z", "3")
         npy = df.AsNumpy(exclude=["z"])
         ref = {"x": np.array([1] * 4), "y": np.array([2] * 4)}
         self.assertTrue(sorted(["x", "y"]) == sorted(npy.keys()))
-        self.assertTrue(any(ref["x"] == npy["x"]))
-        self.assertTrue(any(ref["y"] == npy["y"]))
+        self.assertTrue(all(ref["x"] == npy["x"]))
+        self.assertTrue(all(ref["y"] == npy["y"]))
 
         df2 = ROOT.RDataFrame(4).Define("x", "1").Define("y", "2").Define(
             "z", "3")
         npy = df.AsNumpy(columns=["x", "y"], exclude=["y"])
         ref = {"x": np.array([1] * 4)}
-        self.assertTrue(["x"] == npy.keys())
-        self.assertTrue(any(ref["x"] == npy["x"]))
+        self.assertTrue(sorted(["x"]) == sorted(npy.keys()))
```

Comment:
Isn't `sorted(["x"])` the same as `["x"]`? Why do we need the `sorted` here?

Reply:
Actually we don't need it around `["x"]` but around `dict.keys()`, since this is an implicit `tuple(...)` call. I could have put a `tuple(npy.keys())` as well.

Somehow, the comparison `["x"] == dict.keys()` works in some Python versions and in some not (e.g., the build failing the test above).

Reply:
But the `sorted` is needed for the tests with more than one key, since `dict.keys()` is not ordered. I've put it here as well to keep it similar to the other tests.

Reply:
Looks like while in Python2 `dict.keys()` returns a list, in Python3 it returns a view on the keys (`dict_keys` type), so that if the dictionary changes, the view sees those changes.

For this line, I would then just write `self.assertTrue(["x"] == list(npy.keys())`

Reply:
Fixed!